test/integration/kafka_confluent/kafka_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package kafka_confluent
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	kafka_confluent "github.com/cloudevents/sdk-go/protocol/kafka_confluent/v2"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/test"
+)
+
+const (
+	TEST_GROUP_ID = "test_confluent_group_id"
+)
+
+type receiveEvent struct {
+	event cloudevents.Event
+	err   error
+}
+
+func TestSendEvent(t *testing.T) {
+	test.EachEvent(t, test.Events(), func(t *testing.T, eventIn event.Event) {
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		topicName := "test-ce-confluent-" + uuid.New().String()
+		eventIn = test.ConvertEventExtensionsToString(t, eventIn)
+
+		// start a cloudevents receiver client go to receive the event
+		eventChan := make(chan receiveEvent)
+
+		receiverReady := make(chan bool)
+		go func() {
+			p, err := protocolFactory(t, ctx, "", []string{topicName})
+			if err != nil {
+				eventChan <- receiveEvent{err: err}
+				return
+			}
+			defer p.Close(ctx)
+
+			client, err := cloudevents.NewClient(p)
+
+			receiverReady <- true
+			err = client.StartReceiver(ctx, func(event cloudevents.Event) {
+				eventChan <- receiveEvent{event: event}
+			})
+			if err != nil {
+				eventChan <- receiveEvent{err: err}
+			}
+		}()
+
+		<-receiverReady
+
+		// start a cloudevents sender client go to send the event
+		p, err := protocolFactory(t, ctx, topicName, nil)
+		require.NoError(t, err)
+		client, err := cloudevents.NewClient(p)
+		require.NoError(t, err)
+		res := client.Send(ctx, eventIn)
+		require.NoError(t, res)
+
+		// check the received event
+		receivedEvent := <-eventChan
+		require.NoError(t, receivedEvent.err)
+		eventOut := test.ConvertEventExtensionsToString(t, receivedEvent.event)
+
+		// test.AssertEventEquals(t, eventIn, receivedEvent.event)
+		err = test.AllOf(
+			test.HasExactlyAttributesEqualTo(eventIn.Context),
+			test.HasData(eventIn.Data()),
+			test.HasExtensionKeys([]string{kafka_confluent.KafkaPartitionKey, kafka_confluent.KafkaOffsetKey}),
+			test.HasExtension(kafka_confluent.KafkaTopicKey, topicName),
+		)(eventOut)
+		require.NoError(t, err)
+	})
+}
+
+// To start a local environment for testing:
+// docker run --rm --net=host -e ADV_HOST=localhost -e SAMPLEDATA=0 -p 9091:9091 -p 9092:9092 lensesio/fast-data-dev
```

Comment:
question: should we use the `confluent` docker images for integration testing instead? Assuming many will use this binding with `confluent` Kafka (which is Kafka, I understand, however might have some differences when it comes to defaults, connection/auth handling, etc.)

Reply:
I tested both `lensesio/fast-data-dev` and `confluentinc/confluent-local` images. I didn't find any differences in their usage during the test. 

I understand your concerns. However, running the `confluentinc/confluent-local` in the integration workflow might cause issues with the `lensesio/fast-data-dev`. I can start another PR focus on these issues. What do you think of it?